src/gfn/gflownet/base.py

```diff
@@ -70,7 +75,7 @@ def sample_trajectories(self, env: Env, n_samples: int) -> Trajectories:
         return trajectories
 
 
-class TrajectoryBasedGFlowNet(PFBasedGFlowNet):
+class TrajectoryBasedGFlowNet(PFBasedGFlowNet[Trajectories]):
```

Comment:
I'm wondering whether there is a way to remove the `TrajectoryBasedGFlowNet` altogether, now that `GFlowNet[Trajectories]` exist. They seem to serve the same purpose.

Reply:
I am looking into some way to simplify the class hierarchy and I'm not sure that it is possible. I made this little GraphViz of the class hierarchy of the `gfn.gflownet` module and it looks like `TrajectoryBasedGFlowNet` is necessarily sharing methods like `get_pfs_and_pbs` and `get_trajectories_scores` across `TBGFlowNet`, `SubTBGFlowNet`, and `LogPartitionVarianceGflowNet`:

![classes](https://github.com/saleml/torchgfn/assets/927168/c9af04ec-53d0-4b18-93e7-050815aeab36)

As you can see, given that every branch node (ie: non-leaf node) in the class tree has greater than one children, I don't think there are any trivial ways to compress the hierarchy without duplicating some functionality of a branch node into its current children.

That said, of course if you have any advice on that or if there is anything specific you'd like to see happen with this class hierarchy in order to simplify it, I am more than happy to take a crack at it! 🙏

Reply:
Thanks ! Let's keep things as they are then :)